pythonforandroid/recipe.py

```diff
@@ -783,8 +783,13 @@ def get_recipe_env(self, arch=None, with_flags_in_cc=True):
             hppath.append(join(dirname(self.hostpython_location), 'Lib'))
             hppath.append(join(hppath[0], 'site-packages'))
             builddir = join(dirname(self.hostpython_location), 'build')
-            hppath += [join(builddir, d) for d in listdir(builddir)
-                       if isdir(join(builddir, d))]
+            try:
+                listed = listdir(builddir)
+                hppath += [join(builddir, d) for d in listed
+                           if isdir(join(builddir, d))]
+            except OSError as err:
+                debug('%s', err)
```

Comment:
Yes, I got the exception listed in #1031 that a folder was missing, but it didn't seem to be required, and this allowed it to finish.  It has been long enough, that I don't remember the finer details, but this allowed me to proceed.

We could improve the error handling.  But, I've since moved to the ruamel.yaml package (see pull request), so pyyaml is less important for me now.